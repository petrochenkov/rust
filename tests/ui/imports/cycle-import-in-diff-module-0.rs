// https://github.com/rust-lang/rust/pull/124840#issuecomment-2098148587

mod a {
    pub(crate) use crate::S;
    //~^ ERROR unresolved imports `crate::S`, `self::a::S`
    //~| ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
    //~| ERROR cannot determine resolution for the import
}
mod b {
    pub struct S;
}
use self::a::S; //~ ERROR cannot determine resolution for the import
use self::b::*;

fn main() {}
