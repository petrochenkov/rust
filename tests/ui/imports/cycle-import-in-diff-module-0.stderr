error: cannot determine resolution for the import
  --> $DIR/cycle-import-in-diff-module-0.rs:4:20
   |
LL |     pub(crate) use crate::S;
   |                    ^^^^^^^^

error: cannot determine resolution for the import
  --> $DIR/cycle-import-in-diff-module-0.rs:4:20
   |
LL |     pub(crate) use crate::S;
   |                    ^^^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/cycle-import-in-diff-module-0.rs:4:20
   |
LL |     pub(crate) use crate::S;
   |                    ^^^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: cannot determine resolution for the import
  --> $DIR/cycle-import-in-diff-module-0.rs:13:5
   |
LL | use self::a::S;
   |     ^^^^^^^^^^

error[E0432]: unresolved imports `crate::S`, `self::a::S`
  --> $DIR/cycle-import-in-diff-module-0.rs:4:20
   |
LL |     pub(crate) use crate::S;
   |                    ^^^^^^^^
...
LL | use self::a::S;
   |     ^^^^^^^^^^

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0432`.
